### DISCUSSION TITLE
API Token Becomes Invalid After Container Restart

### DISCUSSION THREAD

**Discussion by tormagnehougen** [ckan/ckan#8391] [Q&A]
"I have a CKAN image that I spin up in an OpenShift cluster. Then, I create an API token using the command ckan -c production.ini user token add ckanuser tokenname. The command returns a token that I use in a third-party application to maintain datasets, and it works fine.

The problem arises when I restart the container; the token no longer works."

**Comment by tino097** [comment]
@tormagnehougen https://docs.ckan.org/en/2.10/maintaining/configuration.html?highlight=secret#beaker-session-secret 
as this is set to [fallback](https://github.com/ckan/ckan/blob/dev-v2.10/ckan/lib/api_token.py#L22) when validating tokens 

if you are using 2.11 there is a new config [value](https://docs.ckan.org/en/latest/maintaining/configuration.html#secret-key) 

From my experience, changing the values on container recreation for this values, could be causing the issue

**Comment by FelixBuehler** [comment]
I have the same issue. 
@tormagnehougen Did you solve this in the meantime?
